src/main/java/tempest/GUI/HomePage.java

```diff
@@ -37,14 +42,14 @@ public JPanel getPanel(){
     }
 
     @Override
-    public void actionPerformed(ActionEvent e){
+    public void actionPerformed(ActionEvent e) {
         Object source = e.getSource();
 
-        if(source == addModuleButton){
-            manager.swapCard("addModule");
+        if(source == addModuleButton) {
+            manager.swapCard(manager.getPageName(AddModulePage.class));
         }
-        else if(source == addSessionButton){
-            manager.swapCard("addSession");
+        else if(source == addSessionButton) {
+            manager.swapCard(manager.getPageName(AddSessionPage.class));
```

Comment:
This seems like it would be better suited to a switch statement.

Reply:
Seems like you can't use switch with an object

Reply:
Could you switch on the class name with `source.getClass().getName()`. Does that ugliness balance out with the switch?

Reply:
It does let me switch on that, but that's a string so then I did "case addModuleButton.getName():" but its gives an error of 'constant expression required'

There's only 2 things, isn't using an if statement fine?